src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -233,6 +287,181 @@ public void promoteWith(RenderedShape<?> rshape)
+
+        @Override
+        public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilder builder, double cx, double cy,
+                double cz, float partialTick) {
+            if (shape.a == 0.0)
+                return;
+            if (shape.needf3b && !client.getEntityRenderDispatcher().shouldRenderHitBoxes()){
+                return;
+            }
+            Vec3 v1 = shape.relativiseRender(client.level, shape.pos, partialTick);
+            Camera camera1 = client.gameRenderer.getMainCamera();
+
+            matrices.pushPose();
+            // matrices.setIdentity();
+            if (!isitem)// blocks should use its center as the origin
+                matrices.translate(0.5, 0.5, 0.5);
+            matrices.translate(v1.x - cx, v1.y - cy, v1.z - cz);
+            if (shape.facing == null) {
+                // matrices.method_34425(new Matrix4f(camera1.getRotation()));
+                matrices.mulPose(camera1.rotation());
+                matrices.mulPose(Vector3f.YP.rotationDegrees(180));
+            } else {
+                switch (shape.facing) {
+                    case NORTH:
+                        break;
+                    case SOUTH:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(180));
+                        break;
+                    case EAST:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(270));
+                        break;
+                    case WEST:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(90));
+                        break;
+                    case UP:
+                        matrices.mulPose(Vector3f.XP.rotationDegrees(90));
+                        break;
+                    case DOWN:
+                        matrices.mulPose(Vector3f.XP.rotationDegrees(-90));
+                        break;
+                }
+            }
+            // RenderSystem.scalef(shape.size* 0.0025f, -shape.size*0.0025f,
+            // shape.size*0.0025f);
+            if (shape.tilt != 0.0f)
+                matrices.mulPose(Vector3f.ZP.rotationDegrees(shape.tilt));
+            if (shape.lean != 0.0f)
+                matrices.mulPose(Vector3f.XP.rotationDegrees(shape.lean));
+            if (shape.turn != 0.0f)
+                matrices.mulPose(Vector3f.YP.rotationDegrees(shape.turn));
+            matrices.scale(shape.scale_x, shape.scale_y, shape.scale_z);
+
+            if (!isitem)// blocks should use its center as the origin
+                matrices.translate(-0.5, -0.5, -0.5);
+            // matrices.scale(-1, 1, 1);
+            RenderSystem.depthMask(true);
+            RenderSystem.enableCull();
+            RenderSystem.enableDepthTest();
+
+            // matrices.scale(-1, 1, 1);
+
+            blockPos = new BlockPos(v1);
+            int light;
+            light = LightTexture.pack(
+                    shape.light_fromblock == -999 ? client.level.getBrightness(LightLayer.BLOCK, blockPos)
```

Comment:
detail, nit, but I woudn't use -999 for a specific use, any value outside of 0-15 would need a default. This way the default could be just -1, right?

Reply:
i change it to -1